plugins/foreman_virt_who_configure/0.5/index.md

```diff
@@ -0,0 +1,272 @@
+
+## 3.5 Authentication and authorization
+
+The plugin creates a few roles that can be useful even without using the configuration functionality.
+
+* **Virt-who Manager** - This role contains all permission to manage virt-who configurations, the user needs this role to create/delete/update configurations.
+* **Virt-who Reporter** - This is a role that contains minimal set of permissions for virt-who to upload the report, it can be used if you configure virt-who manually and want to use a user that has locked down account
+* **Virt-who Viewer** - This user can see all configurations including their configuration scripts, which means viewers could still deploy the virt-who instances for existing virt-who configurations. 
+
+Every virt-who configuration creates a service user which is used for virt-who instance authentication. This means every virt-who configuration uses a different account. The service user login is __virt_who_reporter_$id__ and has a randomly generated password. The user account is assigned __Virt-who Reporter__ role. The service user can't login to Foreman and can only be used for virt-who reporting. 
+
+# 4. API
+
+All functionality that is provided by the WebUI can be also achieved by the REST API. 
+
+The following API endpoints are available:
+```
+* :GET, 'foreman_virt_who_configure/api/v2/configs', N_("List of virt-who configurations")
+
+* :GET, "foreman_virt_who_configure/api/v2/configs/:id/", N_("Show a virt-who configuration")
+
+* :GET, "foreman_virt_who_configure/api/v2/configs/:id/deploy_script/", N_("Renders a deploy script for the specified virt-who configuration")
+
+* :POST, 'foreman_virt_who_configure/api/v2/configs', N_("Create a virt-who configuration")
+
+* :PUT, 'foreman_virt_who_configure/api/v2/configs/:id', N_("Update a virt-who configuration")
+
+* :DELETE, 'foreman_virt_who_configure/api/v2/configs/:id', N_("Delete a virt-who configuration")
+```
+To find more information about how to use the available API endpoints, visit `your.foreman.com/apidoc`.
+
+# 5. Migration from existing virt-who setups
+
+If virt-who was already configured manually and sending reports correctly, there's no reason to migrate to this plugin as it mainly helps with the setup. On the other hand, to monitor incoming reports for multiple virt-who instances at one place, the migration can be useful. To minimize the virt-who downtime, we suggest you check for the minimal required virt-who version before you start the migration. The migration consists of the following steps:
+  
+* Create a virt-who configuration with all parameters from manual setup.
+* Removing the old virt-who configuration file.
+* Deploying the new virt-who configuration from the plugin.
+
+Removing the old configuration is pretty straightforward, we suggest making a backup of the config file which is to be removed or modified just in case something goes wrong.
```

Comment:
```suggestion
Removing the old configuration is pretty straightforward. We suggest making a backup of the config file which is to be removed or modified just in case something goes wrong.
```